department-of-veterans-affairs/vets-website
# [Letters] - Update letters so V1 components are V3 components

## Summary

Updated the following files to no longer have uswds='false'
- src/applications/letters/components/AppContent.jsx
- src/applications/letters/components/DownloadLetterLink.jsx
- src/applications/letters/components/NoAddressBanner.jsx
- src/applications/letters/components/systemDownMessage.jsx
- src/applications/letters/containers/DownloadLetters.jsx
- src/applications/letters/containers/LetterList.jsx
- src/applications/letters/containers/Main.jsx
- src/applications/letters/containers/VeteranBenefitSummaryLetter.jsx
- src/applications/letters/utils/helpers.jsx

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#76123

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: Tried to get screenshots but was unable to repro certain things in localhost/staging

|         | Before | After |
| ------- | ------ | ----- |
| NoAddressBanner.jsx va-alert  | ![Screenshot 2024-03-18 at 3 03 00 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/e3995caa-0e5b-4de3-a067-0014d27eabaa) | ![Screenshot 2024-03-18 at 3 02 31 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/7808514c-3301-462b-bffb-75308e6fc5ca) |
| DownloadLetters.jsx va-segmented-progress-bar  | ![Screenshot 2024-03-18 at 3 10 05 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/3aa86ec8-b348-4833-b506-4bc9acd76afe) | ![Screenshot 2024-03-18 at 3 10 29 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/f977a4f6-853e-45b7-9517-8c759a3dc0b9) |
| LetterList.jsx va-accordion | ![Screenshot 2024-03-18 at 3 22 39 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/dc103f45-5e9c-40da-b605-f97d8d981238) | ![Screenshot 2024-03-18 at 3 48 57 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/dc3f031d-de01-4a41-b074-e3d18b124083) |
| LetterList.jsx va-telephone | ![Screenshot 2024-03-18 at 3 23 07 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/14184576-5183-48a4-bc90-1641e614832e) | ![Screenshot 2024-03-18 at 3 39 59 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/cccba4f0-ed09-4922-9803-ff45486c7513) |
| DownloadLetterLink.jsx va-alert success | ![Screenshot 2024-03-18 at 3 26 11 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/e66f0843-49ba-4ca0-9599-e7891e6046a7) | ![Screenshot 2024-03-18 at 3 49 25 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/694fd268-e10a-4a58-ae08-99c7b117e3ad) |
| Systemdownmessage.jsx va-alert | ![Screenshot 2024-03-18 at 3 41 19 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/eae666f2-3f8e-4e6f-8d34-2ea732a8b314) | ![Screenshot 2024-03-18 at 3 42 37 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/c1d55487-cbda-42e7-b0a7-663af545ba75) |
| helpers va-alert | ![Screenshot 2024-03-18 at 4 07 06 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/06fa0932-4a45-4be6-b372-df735c040c9f) | ![Screenshot 2024-03-18 at 3 56 20 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/c6dc5265-a2e7-47e7-87c2-b857ee863155) |
| VeteranBenefitSummaryLetter.jsx va-table | ![Screenshot 2024-03-18 at 3 53 18 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/c27eff37-607a-4d74-9b2b-6229b39f9a5e) |       |
| AppContent.jsx va-alert | ![Screenshot 2024-03-18 at 3 59 04 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/141954992/8796ecab-86cf-407d-a6ff-c945260e8426) |       |


## What areas of the site does it impact?

Letters

## Acceptance criteria

- [ ] Any components that were updated as USWDS false that are not V3 components should have that prop removed.
- [ ] Design and engineering test locally to confirm design fidelity of the migration before changes are merged

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback
How to test things
- AppContent.jsx , go to user 30 https://staging.va.gov/records/download-va-letters/letters/
- DownloadLetters.jsx va-segmented-progress-bar, go to user 46 https://staging.va.gov/records/download-va-letters/letters/confirm-address
- LetterList.jsx va-accordion, va-telephone, go to user 46 https://staging.va.gov/records/download-va-letters/letters/letter-list
- DownloadLetterLink.jsx va-alert success, go to user 46 and download a letter from https://staging.va.gov/records/download-va-letters/letters/letter-list
- Systemdownmessage.jsx va-alert, unable to repo in staging
- helpers va-alert, go to user 46 and open the Service Verification Letter https://staging.va.gov/records/download-va-letters/letters/letter-list
- VeteranBenefitSummaryLetter.jsx va-table, go to user 46 and open the Benefit Summary and Service Verification Letter https://staging.va.gov/records/download-va-letters/letters/letter-list
